### PR TITLE
Remove sd optional kwarg from distributions

### LINKED ISSUE #5576: Remove sd optional kwarg from distributions
Several continuous univariate distributions have an old undocumented alternative `sd` kwarg, which should be removed in favor of `sigma`

This was brought up in #5546 and discussed internally in Slack

### PR DESCRIPTION
**Pull request by purna135**
Closes #5576 

Removed `sd` kwarg in favor of `sigma` from several continuous univariate distributions.

### CODE REVIEW

**Comment by purna135** [comment]
Should I replace `sd` with `sigma` in the test which are calling any function with `sd` arguments?

**Comment by canyon289** [comment]
> Should I replace `sd` with `sigma` in the test which are calling any function with `sd` arguments?

Yes please, totally eliminate it from the codebase

**Comment by purna135** [comment]
> totally eliminate it from the codebase

Do you mean, the complete test function which are using `sd` parameter ?

**Comment by ricardoV94** [comment]
> > totally eliminate it from the codebase
> 
> Do you mean, the complete test function which are using `sd` parameter ?

No. Keep the function, just rename it to `sigma` as you first suggested.

**Comment by purna135** [comment]
Hello, I'm having trouble understanding what these lines below are doing:

https://github.com/pymc-devs/pymc/blob/5eaa516476d5135ec5df251209a0d188acb55fc2/pymc/tests/test_sampling.py#L737-L739

https://github.com/pymc-devs/pymc/blob/5eaa516476d5135ec5df251209a0d188acb55fc2/pymc/tests/test_transforms.py#L335-L340

**Comment by ricardoV94** [comment]
> Hello, I'm having trouble understanding what these lines below are doing:
> 
> https://github.com/pymc-devs/pymc/blob/5eaa516476d5135ec5df251209a0d188acb55fc2/pymc/tests/test_sampling.py#L737-L739

That one does not seem to involve the sd kwarg. Is it failing or are you asking because the variable has `sd` in the name? If it's passing you can leave as is.

> https://github.com/pymc-devs/pymc/blob/5eaa516476d5135ec5df251209a0d188acb55fc2/pymc/tests/test_transforms.py#L335-L340

That one you need to change that dict to have`sigma` as a key instead of `sd`. I guess in the function that's called the dictionary is unpacked as kwargs like `Halfnormal(**{"sigma": 2})`

**Comment by purna135** [comment]
Hi @ricardoV94, @canyon289,
 
Running pytest takes an excessive amount of time.
I ran the pytest 24 hours ago and it's still at 98 percent.

Is it possible to do a quick test on local system before pushing to github to see if there are any errors?

**Comment by ricardoV94** [comment]
> Hi @ricardoV94, @canyon289,
> 
> Running pytest takes an excessive amount of time. I ran the pytest 24 hours ago and it's still at 98 percent.
> 
> Is it possible to do a quick test on local system before pushing to github to see if there are any errors?

You shouldn't try to run the entire test suite on your local machine. Usually only tests that you added or that you know are directly related to changes you've made.

**Comment by purna135** [comment]
Hi @ricardoV94, I only fixed the errors I was getting the previous time, and it passed all of the tests this time.

However, I can still find some tests/methods/notebooks that use `sd`, for example:

https://github.com/pymc-devs/pymc/blob/8cb71a5b1104166127e89a25d80c34774e182712/pymc/tests/test_shape_handling.py#L392-L397

https://github.com/pymc-devs/pymc/blob/main/docs/source/learn/examples/dimensionality.ipynb
Screenshot from 2022-03-14 17-38-21


**Should I rename them to sigma as well ?**

**Comment by ricardoV94** [comment]
> Should I rename them to sigma as well ?

Yes 👍

**Comment by purna135** [comment]
Hello, @ricardoV94! I believe I am finished; please take a look.
How to resolve this conflicts ?

**Review by twiecki — APPROVED** [review]

**Comment by twiecki** [comment]
Release notes

**Comment by purna135** [comment]
> Release notes

Yes, the release note mentions `sd` but I assumed it would be handled by the Member. Is it possible for me to make changes to that?

**Comment by twiecki** [comment]
Yes certainly.

**Comment by purna135** [comment]
Under the Deprecations header for PyMC vNext, I'll add a point like "In favour of sigma, the deprecated sd has been removed.", right?

**Comment by twiecki** [comment]
Yes, exactly.

**Comment by purna135** [comment]
Done 😃, I believe it's finished now.

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -96,6 +96,7 @@ All of the above apply to:
 This includes API changes we did not warn about since at least `3.11.0` (2021-01).
 
 - Setting initial values through `pm.Distribution(testval=...)` is now `pm.Distribution(initval=...)`.
+- In favour of `sigma`, the deprecated `sd` has been removed.
```
Comment (ricardoV94) [inline RELEASE-NOTES.md]:
```suggestion
- Alternative `sd` keyword argument has been removed from all distributions. `sigma` should be used instead
```

Comment (ricardoV94) [reply RELEASE-NOTES.md]: Should add PR link after description. You can see how that's done in other entries of the release notes

Comment (purna135) [reply RELEASE-NOTES.md]: Done, thanks : )

**Review by ricardoV94 — APPROVED** [review]

**Comment by purna135** [comment]
Is there anything I can do to resolve this conflict?

**Comment by ricardoV94** [comment]
> Is there anything I can do to resolve this conflict?

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-using-the-command-line

**Comment by ricardoV94** [comment]
Thanks @purna135!

**Comment by purna135** [comment]
Thanks for this opportunity ☺️

**Pull request merged**